data/posts/fog-of-war.md

```diff
@@ -0,0 +1,206 @@
+---
+title: Fog of War
+summary: Introducing Fog of War - infinitely scalable Remix and React Router applications
+featured: false
+date: 2024-07-10
+image: /blog-images/headers/fog-of-war.png
+imageAlt: Fog of War
+imageDisableOverlay: true
+authors:
+  - Matt Brophy
+---
+
+Remix is designed to make your application performant by default. Our latest feature, [Fog of War][remix-fog-of-war][^1], ensures your application stays performant no matter how much it grows.
+
+[^1]: The Fog of War feature was released behind an `unstable` flag in Remix [v2.10][remix-2-10] for early beta testing—we hope to stabilize it in an upcoming release
+
+## How Remix makes fetch(es) happen
+
+Remix has primarily been a compiler and server-runtime on top of React Router aimed at giving you the idiomatic and performant way we would have written a React Router SSR app. Could you build your own React Router SSR application without using Remix? Absolutely! However, to get the same kind of performance optimizations, you'd very likely end up writing your own compiler and server-runtime, mimicking a lot of the optimizations Remix has built-in.
+
+Most of these Remix optimizations share a common goal of eliminating network waterfalls.
+
+In order to avoid "render then fetch" waterfalls, Remix [decouples rendering from fetching][when-to-fetch] (see also: [Remixing React Router][remixing-rr]). In order to do this, Remix needs to know your route tree up front so it can kick off the data fetches and download the route modules in parallel whenever a link is clicked. This results in the inverted and more performant approach of "fetch then render" (or "fetch while render" if you're [streaming data][streaming]).
+
+In a "render then fetch" world, your application needs to download the route implementation, then render the component, which would kick off the data fetches - causing a waterfall:
```

Comment:
```suggestion
In a "render then fetch" world your application needs to download the route implementation, then render the component, which would kick off data fetching and cause a waterfall.
```

Reply:
I had reworded this slightly above to clarify based on Brook's feedback - does this new wording seem clear to you?

```
In a "render then fetch" world, your application downloads the route implementation, then kicks off the data fetches while rendering the component—causing a waterfall:
```

Reply:
Sounds good to me! Thanks Broph Dawg